src/bindfs.c

```diff
@@ -1511,11 +1701,24 @@ static int bindfs_open(const char *path, struct fuse_file_info *fi)
 {
     int fd;
     char *real_path;
+    int access_check_result;
+    int path_search_permission;
 
     real_path = process_path(path, true);
     if (real_path == NULL)
         return -errno;
 
+    access_check_result = dirname_access_check(real_path, W_OK);
+    if (access_check_result != 0)
+        return access_check_result;
+
+    path_search_permission = check_path_search_perms(real_path);
+    if (path_search_permission != 0)
+        return path_search_permission;
+
+    // TODO: Pretty sure we need more access checks here. Will let the tests
+    // highlight the issues.
```

Comment:
I suspect something like

```c
bool write_access = (fi->flags & O_WRONLY) || (fi->flags & O_RDWR) || (fi->flags & O_APPEND);
... access_check(real_path, write_access ? W_OK : R_OK) ...
```

Reply:
The below is from `open(2)` when it says  `EACCES` is returned. The emboldened parts are what I think we'd still be missing after incorporating your snippet. That second paragraph could be fun...

> The requested access to the file is not allowed, or search permission is denied for one of the directories in the path prefix of pathname, or **the file did not exist yet and write access to the parent directory  is  not allowed**.  (See also path_resolution(7).)
> 
> **Where  O_CREAT  is  specified,  the protected_fifos or protected_regular sysctl is enabled, the file already exists and  is  a  FIFO  or  regular file, the owner of the file is neither the current user nor the owner of the containing directory, and the containing directory is both world- or group-writable  and  sticky.**   For  details,  see  the  descriptions  of /proc/sys/fs/protected_fifos   and   /proc/sys/fs/protected_regular   in proc(5).

Reply:
I suspect FUSE routes all `O_CREAT` cases to `bindfs_create`, but you're right, that one will be more complicated.

Reply:
👍 I'll mark this one resolved for now if that suits, just so it's off my immediate to-do list; the tests should flag this up again.